Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp

```diff
@@ -477,10 +477,11 @@ void PageClientImpl::wheelEventWasNotHandledByWebCore(const NativeWebWheelEvent&
     if (gdk_event_get_event_type(event.nativeEvent()) != GDK_SCROLL)
         return;
 
-    webkitWebViewBaseForwardNextWheelEvent(WEBKIT_WEB_VIEW_BASE(m_viewWidget));
+    webkitWebViewBaseForwardNextWheelEvent(WEBKIT_WEB_VIEW_BASE(m_viewWidget), event.nativeEvent());
 
 #if USE(GTK4)
-    gdk_display_put_event(gtk_widget_get_display(m_viewWidget), event.nativeEvent());
+    // Note: the docs for gdk_display_put_event lie - this adds to the end of the queue, not the front.
```

Comment:
We need to make sure there is a GTK issue report.

Reply:
Didn't see one, so I filed https://gitlab.gnome.org/GNOME/gtk/-/issues/5525